### ISSUE TITLE
Do you by any chance have example code for sending a message?

### ISSUE THREAD

**Issue by marks** [mkalkbrenner/HVClientLibPHP#2]
Seems to be quite a complex process. Do you have any code or pointers on how to use `HVClientLibPHP` for sending a message to a HV user?

**Comment by mkalkbrenner** [comment]
https://platform.healthvault-ppe.com/platform/XSD/method-sendinsecuremessage.xsd
https://platform.healthvault-ppe.com/platform/XSD/method-sendinsecuremessagefromapplication.xsd

Do you mean one of these methods?

**Comment by marks** [comment]
@mkalkbrenner - I mean http://developer.healthvault.com/pages/types/type.aspx?id=72dc49e1-1486-4634-b651-ef560ed051e5 (sample = http://blogs.msdn.com/cfs-file.ashx/__key/communityserver-blogs-components-weblogfiles/00-00-01-01-96/5700.sample_2D00_message_2D00_thing.xml) I understand the sample XML pretty well but I'm not completely understanding how the blobs work. Is that implemented in this lib already?

Thanks in advance!

**Comment by mkalkbrenner** [comment]
Unfortunately blobs are not yet supported by this lib. Blobs are maybe the most complex part of HealthVault. Read this post by Eric Gunnerson:
http://blogs.msdn.com/b/ericgu/archive/2011/05/24/dealing-with-blob-data-in-healthvault-xml.aspx

Personally I used the Java lib for reverse engineering instead of the .net API because I know Java. I did so more than one year ago. That time there was no support for Blobs. I#m not sure if that has changed:
http://healthvaultjavalib.codeplex.com/discussions/288014
http://healthvaultjavalib.codeplex.com/workitem/10515

But it seems that Alistair made some progress:
http://social.msdn.microsoft.com/Forums/en-US/healthvault/thread/6a13a854-b8d2-4652-9d08-2487e6c3f15d

Compared to my light wight approach he's working on a heavy wight but full featured SDK. Maybe we should sniff the communication with his implementation ...

**Comment by marks** [comment]
Yeah.. Seems like the one thing I want to do is the hardest thing to do with the platform (go figure). I'm pretty committed to figuring it out and am best at ruby/PHP so let me know if and how I can help 

## 

Mark Silverberg
marksilver.net/LinkedIn
(Sent from my iPhone)
